jenkins/automation/rgw/ceph/nightly-ceph-build-deploy-vm-docker.groovy

```diff
@@ -0,0 +1,133 @@
+pipeline {
+    agent {
+        node {
+            label 'docker-centos-7.9.2009-node'
+        }
+    }
+
+    triggers { cron('30 19 * * *') }
+
+    options {
+        timeout(time: 240, unit: 'MINUTES')
+        timestamps()
+        buildDiscarder(logRotator(daysToKeepStr: '30', numToKeepStr: '30'))
+        ansiColor('xterm')
+    }
+
+    parameters {
+        string(name: 'CORTX_RE_REPO', defaultValue: 'https://github.com/Seagate/cortx-re/', description: 'Repository for Cluster Setup scripts.', trim: true)
+        string(name: 'CORTX_RE_BRANCH', defaultValue: 'main', description: 'Branch or GitHash for Cluster Setup scripts.', trim: true)
+    // Please configure docker_host, vm_hosts and OSD_Disks parameter in Jenkins job configuration.
+    }
+
+    stages {
+        stage ('Build Ceph Packages') {
+            steps {
+                script { build_stage = env.STAGE_NAME }
+                script {
+
+                    try {
+                        def buildCeph = build job: 'Ceph Build in Docker', wait: true,
+                        parameters: [
+                            string(name: 'CORTX_RE_BRANCH', value: "${CORTX_RE_BRANCH}"),
+                            string(name: 'CORTX_RE_REPO', value: "${CORTX_RE_REPO}")
+                        ]
+                    }
+
+                    catch(err) {
+                        build_stage = env.STAGE_NAME
+                        error "Failed to Build Ceph Packages."
+                    }
+                }
+            }
+        }
+
+        stage ('Build Ceph Image') {
+            steps {
+                script { build_stage = env.STAGE_NAME }
+                script {
+
+                    try {
+                        def buildCephImage = build job: 'Ceph Build Container Image', wait: true
+                    }
+
+                    catch(err) {
+                        build_stage = env.STAGE_NAME
+                        error "Failed to Build Ceph Image."
+                    }
+                }
+            }
+        }
+
+        stage ('Deploy Ceph on hosts') {
+            steps {
+                script { build_stage = env.STAGE_NAME }
+                script {
+
+                    try {
+                        def deployCephVM = build job: 'Ceph VM Deploy', wait: true,
```

Comment:
please take care of calling job path. change it to original job path if it is going to call job resides in other folder after merge.
If called job is going to be in same folder, avoid this comment

Reply:
Done